pkg/minikube/machine/fix.go

```diff
@@ -94,6 +99,22 @@ func fixHost(api libmachine.API, cc *config.ClusterConfig, n *config.Node) (*hos
 	return h, nil
 }
 
+// warnAboutMisconfiguredMachineConfigPaths will warn users if paths in machine/config.json are misconfigured.
+// This problem may be caused if the minikube home folder is moved to another location https://github.com/kubernetes/minikube/issues/14466
+func warnAboutMisconfiguredMachineConfigPaths(h *host.Host) {
+	var driver map[string]interface{}
+	err := json.Unmarshal(h.RawDriver, &driver)
+	if err != nil {
+		return
+	}
+
+	if h.HostOptions.AuthOptions.StorePath != localpath.MiniPath() || driver["StorePath"] != localpath.MiniPath() {
+		out.WarningT(`Your machine config likely has misconfigured paths. To update your config run the following:
+
+	'minikube config update -p <profile_name>'`)
```

Comment:
I think since the profile can easily be passed into the function similar to line 74, we can use the actual profile value to make it easier on the user.

ie.
```
minikube config update -p %s
```

Reply:
Done.